### PR TITLE
Add scripts to collect power management data

### PR DESCRIPTION
**Pull request by spaceotter**
This contains the code to analyze traces with power management data. This will only work on internal machines with rocmProfileData_smutrace installed.

### CODE REVIEW

File: power_trace.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+set -xe
+
+tmpdir=$(mktemp -d -t "traceXXXXXX")
```
Comment (kuhar) [inline power_trace.sh:5]:
```suggestion
readonly tmpdir="$(mktemp -d -t "traceXXXXXX")"
```

File: power_trace.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+cleanup () {
+    rm -r $tmpdir
```
Comment (kuhar) [inline power_trace.sh:8]:
```suggestion
    rm -r "$tmpdir"
```

File: power_trace.sh [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+}
+trap cleanup EXIT
+
+script_dir=$(dirname $(realpath $0))
```
Comment (kuhar) [inline power_trace.sh:12]:
```suggestion
readonly script_dir="$(cd -- "$(dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd)"
```

File: power_trace.sh [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+real_iree=$(command -v iree-benchmark-module)
```
Comment (kuhar) [inline power_trace.sh:14]:
```suggestion
real_iree="$(command -v iree-benchmark-module)"
```

File: power_trace.sh [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+
+cat << EOF > $tmpdir/iree-benchmark-module
```
Comment (kuhar) [inline power_trace.sh:17]:
```suggestion
cat << EOF > "$tmpdir/iree-benchmark-module"
```

File: power_trace.sh [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+#!/bin/bash
+sudo LD_LIBRARY_PATH=/usr/local/lib runTracer.sh $real_iree \$@
```
Comment (kuhar) [inline power_trace.sh:19]:
```suggestion
sudo LD_LIBRARY_PATH=/usr/local/lib runTracer.sh "$real_iree" "$@"
```

File: power_trace.sh [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+ret=$?
+python3 $script_dir/corellator.py trace.rpd || echo "Returned $?"
```
Comment (kuhar) [inline power_trace.sh:21]:
```suggestion
python3 "$script_dir/corellator.py" trace.rpd || echo "Returned $?"
```

File: power_trace.sh [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+exit $ret
+EOF
+
+chmod a+x $tmpdir/iree-benchmark-module
```
Comment (kuhar) [inline power_trace.sh:25]:
```suggestion
chmod a+x "$tmpdir/iree-benchmark-module"
```

File: power_trace.sh [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+
+export PATH=$tmpdir:$PATH
```
Comment (kuhar) [inline power_trace.sh:27]:
```suggestion
export PATH="$tmpdir:$PATH"
```

File: power_trace.sh [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+
+source $@
```
Comment (kuhar) [inline power_trace.sh:29]:
```suggestion
source "$@"
```

**Review by kuhar — COMMENTED** [review]
Mostly nits for path handling.

Could you explain why create a wrapper `iree-benchmark-module` script instead of invoking the user command directly?

**Comment by spaceotter** [comment]
I'm not sure you'd invoke the user command directly? It wouldn't run with the tracing enabled.

**Pull request merged**